### PR TITLE
Add more functions.

### PR DESCRIPTION
**Pull request by Jakio815**
# Added functions #
## 1. ```return_decrypted_buf()``` ##
Returns the buffer pointer of the decrypted buffer.
If the user gives the read buffer as input, it will return the decrypted buffer.

## 2. ```receive_thread_read_one_each()``` ##
Creates a thread to receive messages, by reading one bytes each at the SST header.

### 2-1. ```read_header_return_data_buf_pointer()``` ###
Reads the SST header, and returns the message type, start pointer of the SST's payload, and the payload's length.

### 2-2. ```read_variable_length_one_byte_each()``` ###
Reads the variable length header recursively one byte each. The ```read()``` function reads one byte each, and returns the variable length buffer's size.

These APIs are for reading one byte each and parsing the data buffer. The ```parse_received_message()``` function does this in once, but the new functions are used when reading one byte each is needed.

# Fix multiple typos and formatting. #

Multiply typos and formatting has been updated.

### CODE REVIEW

File: c_api.c [hunk 1]
```diff
@@ -180,7 +181,8 @@ SST_session_ctx_t *server_secure_comm_setup(
     }
     if (entity_server_state == HANDSHAKE_2_SENT) {
         unsigned char received_buf[MAX_HS_BUF_LENGTH];
-        int received_buf_length = read(clnt_sock, received_buf, HANDSHAKE_3_LENGTH);
+        int received_buf_length =
+            read(clnt_sock, received_buf, HANDSHAKE_3_LENGTH);
```
Comment (Jakio815) [inline c_api.c:185]: Only refactoring.

File: c_api.c [hunk 2]
```diff
@@ -115,7 +115,8 @@ SST_session_ctx_t *server_secure_comm_setup(
 
     if (entity_server_state == IDLE) {
         unsigned char received_buf[MAX_HS_BUF_LENGTH];
-        int received_buf_length = read(clnt_sock, received_buf, HANDSHAKE_1_LENGTH);
+        int received_buf_length =
+            read(clnt_sock, received_buf, HANDSHAKE_1_LENGTH);
```
Comment (Jakio815) [inline c_api.c:119]: Only refactoring.

File: c_api.c [hunk 3]
```diff
@@ -77,7 +77,7 @@ SST_session_ctx_t *secure_connect_to_server(session_key_t *s_key,
         received_buf, received_buf_length, &message_type, &data_buf_length);
     if (message_type == SKEY_HANDSHAKE_2) {
         if (entity_client_state != HANDSHAKE_1_SENT) {
-            printf(
+            error_handling(
                 "Comm init failed: wrong sequence of handshake, "
                 "disconnecting...\n");
         }
```
Comment (Jakio815) [inline c_api.c:83]: Only refactoring.

File: c_common.c [hunk 4]
```diff
@@ -8,7 +8,7 @@ void error_handling(char *message) {
 
 void print_buf(unsigned char *buf, size_t size) {
     char hex[size * 3 + 1];
-    for(size_t i = 0; i < size; i++) {
+    for (size_t i = 0; i < size; i++) {
```
Comment (Jakio815) [inline c_common.c:11]: Only refactoring.

File: c_common.c [hunk 5]
```diff
@@ -36,8 +36,7 @@ unsigned int read_unsigned_int_BE(unsigned char *buf, int byte_length) {
     return num;
 }
 
-uint64_t read_unsigned_long_int_BE(unsigned char *buf,
-                                            int byte_length) {
+uint64_t read_unsigned_long_int_BE(unsigned char *buf, int byte_length) {
```
Comment (Jakio815) [inline c_common.c:39]: Only refactoring.

File: c_common.c [hunk 6]
```diff
@@ -47,8 +46,7 @@ uint64_t read_unsigned_long_int_BE(unsigned char *buf,
 }
 
 void var_length_int_to_num(unsigned char *buf, unsigned int buf_length,
-                           unsigned int *num,
-                           unsigned int *var_len_int_buf_size) {
+                           unsigned int *num, uint16_t *var_len_int_buf_size) {
```
Comment (Jakio815) [inline c_common.c:49]: Only reformatting.

File: c_common.c [hunk 7]
```diff
@@ -76,11 +74,47 @@ unsigned char *parse_received_message(unsigned char *received_buf,
                                       unsigned char *message_type,
                                       unsigned int *data_buf_length) {
     *message_type = received_buf[0];
-    unsigned int payload_buf_length;
+    unsigned int var_length_buf_size;
```
Comment (Jakio815) [inline c_common.c:77]: Only changing name.

File: c_secure_comm.h [hunk 8]
```diff
@@ -15,7 +15,7 @@
 #define IN_COMM 30
 #define SESSION_KEY_EXPIRATION_TIME_SIZE 6
 #define HANDSHAKE_1_LENGTH 74
-#define HANDSHAKE_3_LENGTH 82 
+#define HANDSHAKE_3_LENGTH 82
```
Comment (Jakio815) [inline c_secure_comm.h:18]: Only reformatting.

File: c_secure_comm.h [hunk 9]
```diff
@@ -161,9 +161,18 @@ unsigned char *check_handshake_2_send_handshake_3(unsigned char *data_buf,
 // @param data input data buffer
 // @param data_length length of data buffer
 // @param SST_session_ctx_t session ctx struct
-void print_recevied_message(unsigned char *data, unsigned int data_length,
+void print_received_message(unsigned char *data, unsigned int data_length,
```
Comment (Jakio815) [inline c_secure_comm.h:164]: Only reformatting.

File: c_api.c [hunk 10]
```diff
@@ -246,7 +265,19 @@ void receive_message(unsigned char *received_buf,
     unsigned char *data_buf = parse_received_message(
         received_buf, received_buf_length, &message_type, &data_buf_length);
     if (message_type == SECURE_COMM_MSG) {
-        print_recevied_message(data_buf, data_buf_length, session_ctx);
+        print_received_message(data_buf, data_buf_length, session_ctx);
```
Comment (Jakio815) [inline c_api.c:268]: Fix typos.

**Review by hokeun — APPROVED** [review]
Looks good to me.

**Pull request merged**